app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEffectHandler.kt

```diff
@@ -84,10 +89,20 @@ class ScheduleAppointmentEffectHandler @AssistedInject constructor(
 
   private fun loadAppointmentFacility(): ObservableTransformer<LoadAppointmentFacility, ScheduleAppointmentEvent> {
     return ObservableTransformer { effects ->
-      effects.map { AppointmentFacilityLoaded(currentFacility.get()) }
+      effects
+          .observeOn(schedulers.io())
+          .map { patientRepository.patientImmediate(it.patientUuid) }
+          .map { getAssignedFacility(it).toNullable() ?: currentFacility.get() }
```

Comment:
While we know that this is temporary, we have no idea how long it might be temporary for. In reality, it might be months or even more than a year till we reach the point where we can remove this.

I think we should load both the current facility and the patient's assigned facility in this screen and decide in `Update` which one to use.

Reply:
Done